Samples/MapScenarios/MapScenarios/Scenarios/WalkTimeViewController.swift

```diff
@@ -23,51 +23,139 @@ class WalkTimeViewController: UIViewController {
+                } else {
+                    self?.startManagedLocationManager()
                 }
             })
         }
     }
     
     func startManagedLocationManager() {
-        let managedLocationManager = PWManagedLocationManager(buildingId: buildingIdentifier)
         DispatchQueue.main.async { [weak self] in
+            guard let buildingIdentifier = self?.buildingIdentifier else {
+                return
+            }
+            let managedLocationManager = PWManagedLocationManager(buildingId: buildingIdentifier)
             self?.mapView.register(managedLocationManager)
         }
     }
     
     func configureMapViewConstraints() {
         mapView.translatesAutoresizingMaskIntoConstraints = false
-        mapView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
+        mapView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
         mapView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
         mapView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
         mapView.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
     }
+    
+    func configureWalkTimeLabel() {
+        if walkTimeLabel.superview == nil {
+            view.addSubview(walkTimeLabel)
+            walkTimeLabel.translatesAutoresizingMaskIntoConstraints = false
+            walkTimeLabel.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+            walkTimeLabel.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 10.0).isActive = true
+            walkTimeLabel.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: 10.0).isActive = true
+            walkTimeLabel.heightAnchor.constraint(equalToConstant: 50.0).isActive = true
+        }
+    }
+    
+    func updateWalkTimeLabel() {
+        guard let route = mapView.currentRoute, let currentInstruction = mapView.currentRouteInstruction(), var instructionIndex = route.routeInstructions.index(of: currentInstruction) else {
+            walkTimeLabel.text = ""
+            return
+        }
+        // Use user location for current route instruction index if possible
+        if let userLocation = mapView.indoorUserLocation, let closestInstruction = route.closestInstructionTo(userLocation), let closestInstructionIndex = route.routeInstructions.index(of: closestInstruction) {
+            instructionIndex = closestInstructionIndex
+        }
+        
+        let estimatedWalkTimeInMinutes = calculateEstimatedWalkTimeInMinutes(route: route, currentRouteInstructionIndex: instructionIndex)
+        if estimatedWalkTimeInMinutes <= 1 {
+            walkTimeLabel.text = "Under 1 minute til arrival"
+        } else {
+            walkTimeLabel.text = "\(estimatedWalkTimeInMinutes) minutes til arrival"
+        }
+    }
+    
+    func calculateEstimatedWalkTimeInMinutes(route: PWRoute, currentRouteInstructionIndex: Int) -> Int {
```

Comment:
This is just my personal preference but I hate putting all this logic in a view controller. It's just messy. Could this be done with an extension of PWRoute and feeding in the index and estimated walk speed? This way we are showing a cleaner way to extend functionality we give out of the box in a reusable way.